packages/lms-client/src/retrieval/RetrievalOpts.ts

```diff
@@ -138,7 +138,7 @@ export type RetrievalOpts = RetrievalCallbacks & {
 };
 export const retrievalOptsSchema = z.object({
   chunkingMethod: retrievalChunkingMethodSchema.optional(),
-  limit: z.number().optional(),
+  limit: z.number().int().optional(),
```

Comment:
@ryan-the-crayon This is the one I'm least sure about, as I believe this is the limit on the amount of data to download, but if it's a limit on the download data rate then presumably it should remain a floating point value.

Reply:
@ncoghlan this is a limit on the number of results to return, so the change here is good